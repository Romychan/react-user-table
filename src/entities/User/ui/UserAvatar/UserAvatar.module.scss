@import '~/app/assets/styles/_typography.scss';

.profile {
  display: flex;
  align-items: center;
  width: calc(100%);
}

.information {
  margin-left: 1rem;
  overflow: hidden;
}

.name,
.username {
  text-overflow: ellipsis;
  overflow: hidden;
}

.name {
  @include text-sm;
  color: var(--gray-900);
}

.username {
  @include text-xs;
  color: var(--gray-600);
}
