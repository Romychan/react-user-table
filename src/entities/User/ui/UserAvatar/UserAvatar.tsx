import { Avatar } from '~/shared/ui/Avatar';

import styles from './UserAvatar.module.scss';

interface IUserAvatarProps {
  /** The user's first and last name */
  name: string;
  /** The user's username */
  username: string;
  /** The user's avatar */
  image: string;
}

/** This component is used to display the user's avatar, name and username */
export const UserAvatar = ({ name, username, image }: IUserAvatarProps) => {
  return (
    <div className={styles.profile}>
      <Avatar src={image} name={username} alt={username} size="xs" isBordered />

      <div className={styles.information}>
        <h2 className={styles.name}>{name}</h2>
        <p className={styles.username}>{username}</p>
      </div>
    </div>
  );
};
