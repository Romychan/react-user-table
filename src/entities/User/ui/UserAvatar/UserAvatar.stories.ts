import type { Meta, StoryObj } from '@storybook/react';

import { MOCK_USER } from '../../api/__mocks__/mocks';

import { UserAvatar } from './UserAvatar';

const meta = {
  title: 'Entities/UserAvatar',
  component: UserAvatar,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
} satisfies Meta<typeof UserAvatar>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    image: MOCK_USER.image,
    name: MOCK_USER.name,
    username: MOCK_USER.username,
  },
};
