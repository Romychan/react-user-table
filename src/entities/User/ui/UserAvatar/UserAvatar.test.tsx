import { render, screen } from '@testing-library/react';

import { UserAvatar } from './UserAvatar';

describe('UserAvatar', () => {
  it('should render correctly', () => {
    render(<UserAvatar image="test-src" name="Test" username="testname" />);

    const avatarElement = screen.getByTestId('avatar');
    const nameElement = screen.getByText('Test');
    const usernameElement = screen.getByText('testname');
    const imageElement = screen.getByRole('img');

    expect(avatarElement).toBeInTheDocument();
    expect(nameElement).toBeInTheDocument();
    expect(usernameElement).toBeInTheDocument();
    expect(imageElement.getAttribute('src')).toMatch('test-src');
  });

  it('should match snapshot', () => {
    const container = render(
      <UserAvatar image="test-src" name="Test" username="testname" />,
    );

    expect(container).toMatchSnapshot();
  });
});
