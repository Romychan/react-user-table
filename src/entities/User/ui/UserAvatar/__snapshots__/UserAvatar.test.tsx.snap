// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserAvatar > should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_profile_847f2f"
      >
        <div
          class="_container_15af9b _xs_15af9b _auto_15af9b _border_15af9b"
          data-testid="avatar"
        >
          <img
            alt="testname"
            class="_image_15af9b"
            src="test-src"
          />
        </div>
        <div
          class="_information_847f2f"
        >
          <h2
            class="_name_847f2f"
          >
            Test
          </h2>
          <p
            class="_username_847f2f"
          >
            testname
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_profile_847f2f"
    >
      <div
        class="_container_15af9b _xs_15af9b _auto_15af9b _border_15af9b"
        data-testid="avatar"
      >
        <img
          alt="testname"
          class="_image_15af9b"
          src="test-src"
        />
      </div>
      <div
        class="_information_847f2f"
      >
        <h2
          class="_name_847f2f"
        >
          Test
        </h2>
        <p
          class="_username_847f2f"
        >
          testname
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
